### ISSUE TITLE
Not handled screenshot capture and attaching screenshot in report.html once test execution done and tests failed

### ISSUE THREAD

**Issue by TestMagic0989** [DetachHead/pytest-robotframework#113]
Is there a way to manage the screenshot capture and attachment to the report.html file in pytest_robotframework, especially when tests fail, given the current set of dependencies such as 
`pytest 7.4.2, pytest_robotframework 1.10.0, selenium 4.13.0 or webdriver-manager 4.0.1, and other robotframework-related dependencies.`
 It seems that according to the pytest_robotframework documentation, this functionality is not natively supported, and using external libraries like Allure or pytest-html may not be effective in achieving the desired result since we're aiming to replicate the report.html format provided by robotframework using pytest_robotframework as a pip dependency. Do you have any suggestions or ideas on how to address this, or is it something that may be considered for future development and enhancement?

**Comment by DetachHead** [comment]
hmm, i use a customized screenshot function so i'm not too familiar with the robot's standard screenshot functionality.

do you mind providing an example of how you're taking screenshots that doesn't work with this plugin?

**Comment by TestMagic0989** [comment]
ok, I do not have specific way as i tried before but right now using some workaround in combination on robotframework, So could you please help me by giving here an example how are you handling with screenshot using this pytest_robotframework plugin as well as if you want you test to run in parallel using xdist or parallel pip dep does this plugin pytest_robotframework  support it as well?
thank you!

**Comment by DetachHead** [comment]
> So could you please help me by giving here an example how are you handling with screenshot using this pytest_robotframework

i made a simple example using seleniumlibrary, converting a robot test that takes a screenshot to python like this:
```robot
# before:

*** Settings ***
Library     SeleniumLibrary


*** Test Cases ***
screenshot
    Open Browser    https://google.com    Chrome
    Capture Page Screenshot
```
```py
# after:

from SeleniumLibrary import SeleniumLibrary

def test_screenshot():
    sl = SeleniumLibrary()
    sl.open_browser("https://google.com", "Chrome")
    sl.capture_page_screenshot()
```
which seems to work just fine:
![image](https://github.com/DetachHead/pytest-robotframework/assets/57028336/ad136feb-2bc3-419a-919e-b76f96d1ab68)

does that answer your question?

> if you want your test to run in parallel using `xdist or parallel `pip dep does this plugin pytest_robotframework support it as well?

xdist isn't supported yet. see #98

**Comment by TestMagic0989** [comment]
First of all thank you so very much for answering my question!!
So, it means for screenshot even if we use plugin pytest_robotframework we do have to rely on robotframework SeleniumLibrary explicitly to capture screenshot right? and there is no workaround if in case i do not want to use SeleniumLibrary ? because I was thinking is this plugin pytest_robotframework is supporting us to run test in pytest and gets the exactly same reporting format then we can simply use python based selenium dep and do not rely on robotframework SeleniumLibrary but this isn't the case right?

and for xdist we do not support it yet and when do you think we can start supporting it or do we have any workaround we can run our test in parallel.

**Comment by DetachHead** [comment]
> So, it means for screenshot even if we use plugin pytest_robotframework we do have to rely on robotframework SeleniumLibrary explicitly to capture screenshot right? and there is no workaround if in case i do not want to use SeleniumLibrary ? because I was thinking if this plugin pytest_robotframework is supporting us to run test in pytest and gets the exactly same reporting format then we can simply use python based selenium dep and do not rely on robotframework SeleniumLibrary but this isn't the case right?

ah i get what you're saying now. you're right that it's less necessary to use seleniumlibrary when you're writing your tests in python, because its api is designed for the robot language. however when it comes to reporting to the robot log it can still be useful. but nothing's stopping you from implementing your own screenshot function if you'd rather not use the one in seleniumlibrary. here's an example of what that might look like:
```py
from robot.api.logger import info
from selenium import webdriver
from selenium.webdriver.firefox.webdriver import WebDriver

from pytest_robotframework import keyword

@keyword
def take_screenshot(driver: WebDriver):
    info(
        f'<img src="data:image/png;base64,{driver.get_screenshot_as_base64()}">',
        html=True,
    )

def test_screenshot():
    driver = webdriver.Firefox()
    driver.get("http://www.python.org")
    take_screenshot(driver)
    driver.close()
```
---
> and for xdist we do not support it yet and when do you think we can start supporting it

i'm not sure. i'm not really familiar with how xdist works, but when i attempted to use it with my plugin it caused some strange behavior with the cli arguments. feel free to try it yourself and comment on #98 with any specific issues you encounter.

---

> do we have any workaround we can run our test in parallel.

for running tests in parallel in your CI i can recommend [pytest-shard](https://pypi.org/project/pytest-shard/), which allows you to split the tests into separate jobs to speed up your pipeline.

but for running tests in parallel on your local machine i don't know of any workarounds i'm afraid. since robot itself doesn't allow you to run multiple tests at a time it wasn't something i really considered

**Comment by DetachHead** [comment]
closing this, feel free to comment again or raise another issue if you have any more questions

**Comment by DetachHead** [comment]
> and for xdist we do not support it yet and when do you think we can start supporting it or do we have any workaround we can run our test in parallel.

i got xdist working today. aiming to push an update some time this week
